tests/PHPStan/Analyser/AnalyserIntegrationTest.php

```diff
@@ -859,6 +859,14 @@ public function testBug7275(): void
 		$this->assertNoErrors($errors);
 	}
 
+	public function testBug7484(): void
+	{
+		$errors = $this->runAnalyse(__DIR__ . '/data/bug-7484.php');
+		$this->assertCount(1, $errors);
+		$this->assertSame('Generator expects key type K of int|string, (K of int)|string given.', $errors[0]->getMessage());
+		$this->assertSame(21, $errors[0]->getLine());
```

Comment:
FYI I usually keep AnalyserIntegrationTest for things that crashed PHPStan/caused infinite loop. It's better to put the test in a specific rule test case. (or a specific rule test case + NodeScopeResolverTest for type inference verification).

But I'm gonna merge it like this so that the fix isn't belated :)

Reply:
I'll keep it in mind :)